libs/dao/src/main/java/com/akto/dto/ApiCollectionUsers.java

```diff
@@ -68,8 +68,23 @@ public enum CollectionType {
             });
         }});
 
-    public static Set<String> getUrlsFromConditions(List<CollectionCondition> conditions){
+    public static Set<ApiInfoKey> getApisFromConditions(List<CollectionCondition> conditions) {
         Set<ApiInfoKey> apis = new HashSet<>();
+        
+        // all AND conditions should be last
+        conditions.sort((CollectionCondition c1, CollectionCondition c2) -> {
+            if(c1.getOperator().equals(c2.getOperator())){
+                return 0;
+            } else if(c1.getOperator().equals(CollectionCondition.Operator.OR)) {
+                return -1;
+            }
+            return 1;
+        });
+
+        if(!conditions.isEmpty()){
+            apis.addAll(conditions.get(0).returnApis());
```

Comment:
`get(0)`?

Reply:
since we can have a single AND condition, but when doing AND with conditions, we use retain operator, thus adding the first value explicitly from the list so that the following condition is not a no-op.